Mage/src/main/java/mage/abilities/ActivatedAbility.java

```diff
@@ -7,41 +7,51 @@
 import mage.constants.TimingRule;
 import mage.game.Game;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.UUID;
 
 /**
- * @author BetaSteward_at_googlemail.com
+ * @author BetaSteward_at_googlemail.com, Susucr
  */
 public interface ActivatedAbility extends Ability {
 
     final class ActivationStatus {
 
-        private final boolean canActivate;
-        private final ApprovingObject approvingObject;
+        private final Set<ApprovingObject> approvingObjects;
 
-        public ActivationStatus(boolean canActivate, ApprovingObject approvingObject) {
-            this.canActivate = canActivate;
-            this.approvingObject = approvingObject;
+        // If true, the Activation Status will not check if there is an approvingObject.
+        private final boolean forcedCanActivate;
+
+        public ActivationStatus(ApprovingObject approvingObject) {
+            this(new HashSet<>(Arrays.asList(approvingObject)));
+        }
+
+        public ActivationStatus(Set<ApprovingObject> approvingObjects) {
+            this(false, approvingObjects);
+        }
+
+        private ActivationStatus(boolean forcedCanActivate, Set<ApprovingObject> approvingObjects){
+            this.forcedCanActivate = forcedCanActivate;
+            this.approvingObjects = new HashSet<>();
+            this.approvingObjects.addAll(approvingObjects);
         }
 
         public boolean canActivate() {
-            return canActivate;
+            return forcedCanActivate || !approvingObjects.isEmpty();
         }
 
-        public ApprovingObject getApprovingObject() {
-            return approvingObject;
+        public Set<ApprovingObject> getApprovingObjects() {
+            return approvingObjects;
         }
 
         public static ActivationStatus getFalse() {
-            return new ActivationStatus(false, null);
+            return new ActivationStatus(new HashSet<>());
```

Comment:
again, this seems like it could be an unnecessarily expensive way to initialize. Collections.emptySet() maybe? If never needs to be mutated there might be a way to use a static instance even